magefiles/upgrade/upgrade.go

```diff
@@ -0,0 +1,260 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openshift/oc/pkg/cli/admin/upgrade"
+	"github.com/openshift/oc/pkg/cli/admin/upgrade/channel"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kubeclient "k8s.io/client-go/kubernetes"
+	podUtils "k8s.io/kubectl/pkg/util/podutils"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	k8swait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+var adminAcks = map[string]string{
+	"4.14": "{\"data\":{\"ack-4.13-kube-1.27-api-removals-in-4.14\":\"true\"}}",
```

Comment:
IIUC the admin ack is required only in case there are some deprecated APIs in a new version of OCP
([it was required](https://docs.openshift.com/container-platform/4.9/updating/updating-cluster-prepare.html) for 4.9, 4.12, 4.13, 4.14, but not for 4.10 and 4.11)

However I found a way to detect that requirement easily, and will add it. Thanks for suggestion

Reply:
updated and verified locally:
```
❯ make ci/test/openshift-upgrade
./mage -v ci:performOpenShiftUpgrade
Running target: CI:PerformOpenShiftUpgrade
I0221 13:21:33.830021   63607 upgrade.go:77] current channel is "fast-4.13", current ocp version is "4.13.31"
I0221 13:21:33.965589   63607 upgrade.go:110] desired major.minor version is "4.14", desired channel is "fast-4.14"
I0221 13:21:34.389384   63607 upgrade.go:184] admin ack {"data":{"ack-4.13-kube-1.27-api-removals-in-4.14":"true"}} successfully applied
I0221 13:21:34.526025   63607 upgrade.go:195] found the desired version "4.14.13" in available updates
Requesting update to 4.14.13
I0221 13:21:34.941656   63607 upgrade.go:224] upgrading from 4.13.31 - current progress: Cluster version is 4.13.31
I0221 13:21:35.077201   63607 upgrade.go:224] upgrading from 4.13.31 - current progress: Cluster version is 4.13.31
I0221 13:21:55.216640   63607 upgrade.go:224] upgrading from 4.13.31 - current progress: Cluster version is 4.13.31
I0221 13:22:15.238220   63607 upgrade.go:224] upgrading from 4.13.31 - current progress: Working towards 4.14.13: 9 of 860 done (1% complete)
```